File: app/src/main/java/com/rtukpe/fixtures/ui/competition/table/TablePresenter.java
package com.rtukpe.fixtures.ui.competition.table;

import com.rtukpe.fixtures.data.manager.DataManager;
import com.rtukpe.fixtures.ui.base.BasePresenter;
import com.rtukpe.fixtures.utils.rx.SchedulerProvider;

import javax.inject.Inject;

import io.reactivex.disposables.CompositeDisposable;

public class TablePresenter<V extends TableMvpView> extends BasePresenter<V> implements TableMvpContract<V> {

    @Inject
    public TablePresenter(DataManager dataManager, SchedulerProvider schedulerProvider, CompositeDisposable compositeDisposable) {
        super(dataManager, schedulerProvider, compositeDisposable);
    }

    @Override
    public void onViewInitialized() {
        super.onViewInitialized();
    }

    @Override
    public void getCompetitionsTable() {
        getCompositeDisposable().add(
                getDataManager().getMockCompetitionTable()
                        .subscribeOn(getSchedulerProvider().io())
                        .observeOn(getSchedulerProvider().ui())
                        .subscribe(leagueTable -> getMvpView().updateTableStandings(leagueTable.standing))
        );
    }
}

File: app/src/main/java/com/rtukpe/fixtures/ui/competition/table/TablePresenter.kt
package com.rtukpe.fixtures.ui.competition.table

import com.rtukpe.fixtures.data.manager.DataManager
import com.rtukpe.fixtures.ui.base.BasePresenter
import com.rtukpe.fixtures.utils.rx.SchedulerProvider
import io.reactivex.disposables.CompositeDisposable
import timber.log.Timber
import javax.inject.Inject

class TablePresenter<V : TableMvpView> @Inject
constructor(dataManager: DataManager, schedulerProvider: SchedulerProvider, compositeDisposable: CompositeDisposable) : BasePresenter<V>(dataManager, schedulerProvider, compositeDisposable), TableMvpContract<V> {

    override fun onViewInitialized() {
        super.onViewInitialized()
    }

    override fun getCompetitionsTable(id: Int) {
        compositeDisposable.add(
                dataManager.getCompetitionTable(id)
                        .subscribeOn(schedulerProvider.io())
                        .observeOn(schedulerProvider.ui())
                        .subscribe({ mvpView.updateTableStandings(it.standing) }, { Timber.e(it) })
        )
    }
}
